x/pki/keeper/msg_server_approve_revoke_x_509_root_cert.go

```diff
@@ -49,38 +49,74 @@ func (k msgServer) ApproveRevokeX509RootCert(goCtx context.Context, msg *types.M
 	revocation.Approvals = append(revocation.Approvals, &grant)
 
 	// check if proposed certificate revocation has enough approvals
-	if len(revocation.Approvals) >= k.CertificateApprovalsCount(ctx, k.dclauthKeeper) {
+	if len(revocation.Approvals) >= k.CertificateApprovalsCount(ctx, k.dclauthKeeper) { //nolint:nestif
 		certificates, found := k.GetApprovedCertificates(ctx, msg.Subject, msg.SubjectKeyId)
+		if !found {
+			return nil, pkitypes.NewErrCertificateDoesNotExist(msg.Subject, msg.SubjectKeyId)
+		}
+
+		var certBySerialNumber *types.Certificate
 		// Assign the approvals to the root certificate
 		for _, cert := range certificates.Certs {
 			if cert.IsRoot {
 				cert.Approvals = revocation.Approvals
```

Comment:
It looks like that if revocation approval is granted for a specific certificate with a given serial number, we can also set approval for certificates that have the same subject and Subject Key Identifier (SKID) but with a different serial numbers.